lib/connectors/memory.js

```diff
@@ -250,6 +250,43 @@ Memory.prototype.updateOrCreate = function(model, data, options, callback) {
     }
   });
 };
+Memory.prototype.patchOrCreateWithWhere =
+  Memory.prototype.upsertWithWhere = function(model, where, data, options, callback) {
+    var self = this;
+    var element = Object.keys(where)[0];
+    var filter = { where: where };
+    if (!data.hasOwnProperty(element)) {
+      data[element] = where[element];
+    }
+    var primaryKey = this.idName(model);
+    self.all(model, filter, options, function(err, instances) {
+      if (err) callback(err);
+      else {
+        var modelsLength = instances.length;
+        if (modelsLength === 0 && data.hasOwnProperty(primaryKey)) {
```

Comment:
I find this confusing, why is it required to provide a primary key in data payload?